solr/core/src/java/org/apache/solr/search/neural/KnnQParser.java

```diff
@@ -84,30 +84,20 @@ public Query parse() {
   }
 
   private Query getFilterQuery() throws SolrException {
-    if (!isFilter()) {
+    boolean isSubQuery = recurseCount != 0;
+    if (!isFilter() && !isSubQuery) {
       String[] filterQueries = req.getParams().getParams(CommonParams.FQ);
       if (filterQueries != null && filterQueries.length != 0) {
-        List<Query> filters;
-
         try {
-          filters = QueryUtils.parseFilterQueries(req, true);
-        } catch (SyntaxError e) {
+          List<Query> filters = QueryUtils.parseFilterQueries(req, true);
```

Comment:
Boolean parameters are kind of sadly confusing in methods when it's not obvious what it is in code review tools (because the method name isn't a hint either).  Here, it is "fixNegativeQueries".  It should be rare to need to do that.  It's redundant/needless here because getProcessedFilter deals with the matter.

Reply:
Ok,  now I remember this bit, I was initially (and I am still) not a fan of that boolean param (as I tend to forget what it means and the name also doesn't help).
We discussed for a better name(we couldn't find) and the reason to put it that was was because that piece of code was shared across various Solr parts and other reviewers thought it was better to isolate it and move it to the queryUtils.

Not sure of course, if it's still the best solution to be honest

Reply:
I think this boolean could be removed (logic is to not call makeQueryable inside).  The only spot to be improved is then RealTimeGetComponent.  Simply update the spot that loops over the filters (RealTimeGetComponent.process line ~ 328 to ensure the query is "queryable".

Reply:
done!